### PR TITLE
Fixed RPCServer throwing errors on CORS preflight

### PR DESCRIPTION
**Pull request by CrackDavid**
Signed-off-by: Christian Busch <kontakt@christian-busch.com>

### Proposed changes in this pull request
Some Servers sent a CORS preflight to see if the endpoint can actually understand the CORS-protocol. This is a OPTIONS request that causes the RPCServer to run into an error (https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request). 
Made RPCServer.go only to listen for POST-requests and allowed needed header-requests.

### Type (put an `x` where ever applicable)
- [x] Bug fix
- [ ] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### Checklist
Please put an `x` against the checkboxes. Write a small comment explaining if its `N/A` (not applicable)

- [x] Read the [CONTRIBUTION guidelines](https://github.com/nknorg/nkn#contributing).
- [ ] All the tests are passing after the introduction of new changes.
- [ ] Added tests respective to the part of code I have written.
- [ ] Added proper documentation where ever applicable (in code and README.md).
- [x] Code has been written according to [NKN-Golang-Style-Guide](https://github.com/nknorg/nkn/wiki/NKN-Golang-Style-Guide)

### Extra information
none

### CODE REVIEW

File: api/httpjson/RPCserver.go [hunk 1]
```diff
@@ -60,91 +60,69 @@ func (s *RPCServer) Handle(w http.ResponseWriter, r *http.Request) {
...
+			w.Write(data)
+		} else {
+			//if the function does not exist
+			log.Warn("HTTP JSON RPC Handle - No function to call for ", request["method"])
+			data, err := json.Marshal(map[string]interface{}{
 				"jsonrpc": "2.0",
 				"error": map[string]interface{}{
-					"code":    -errcode,
-					"message": common.ErrMessage[errcode],
+					"code":    -32601,
```
Comment (dreamfly281) [inline api/httpjson/RPCserver.go]: It is better using some meaning errcode instead of the number.

Comment (CrackDavid) [reply api/httpjson/RPCserver.go]: These lines weren't changed from the original file (https://github.com/nknorg/nkn/blob/6a62178497c8e922b4cba2660103064343ade501/api/httpjson/RPCserver.go#L137) - although git tells you so :)
I changed the code according to the responses above so it is more dynamic. Can you please have a look into that?

File: api/httpjson/RPCserver.go [hunk 2]
```diff
@@ -60,91 +60,70 @@ func (s *RPCServer) Handle(w http.ResponseWriter, r *http.Request) {
...
+			}
+			if err != nil {
+				log.Error("HTTP JSON RPC Handle - json.Marshal: ", err)
+				return
+			}
+			w.Write(data)
+		} else {
+			//if the function does not exist
+			log.Warn("HTTP JSON RPC Handle - No function to call for ", request["method"])
+			response := function(s, request["params"].(map[string]interface{}))
```
Comment (yilunzhang) [inline api/httpjson/RPCserver.go]: This line can be removed

File: api/httpjson/RPCserver.go [hunk 3]
```diff
@@ -60,91 +60,70 @@ func (s *RPCServer) Handle(w http.ResponseWriter, r *http.Request) {
...
+			errcode := response["error"].(common.ErrCode)
```
Comment (yilunzhang) [inline api/httpjson/RPCserver.go]: This line can be changed to
errcode := common.INVALID_METHOD

**Pull request merged**